### PR TITLE
Delete some usage of FakeEtcdClient from master_test.go

### PR DESCRIPTION
**Pull request by wojtek-t**
Ref #15349 

@timothysc

### CODE REVIEW

**Comment by wojtek-t** [comment]
FYI - I will remove the remaining references to FakeEtcdClient in this file in the subsequent PR.

File: pkg/master/master_test.go [hunk 1]
```diff
@@ -56,28 +57,31 @@ import (
...
+
 	master := Master{}
 	config := Config{}
-	fakeClient := tools.NewFakeEtcdClient(t)
-	fakeClient.Machines = []string{"http://machine1:4001", "http://machine2", "http://machine3:4003"}
 	storageVersions := make(map[string]string)
 	storageDestinations := NewStorageDestinations()
-	storageDestinations.AddAPIGroup("", etcdstorage.NewEtcdStorage(fakeClient, testapi.Default.Codec(), etcdtest.PathPrefix()))
-	storageDestinations.AddAPIGroup("extensions", etcdstorage.NewEtcdStorage(fakeClient, testapi.Extensions.Codec(), etcdtest.PathPrefix()))
+	storageDestinations.AddAPIGroup(
```
Comment (timothysc) [inline pkg/master/master_test.go]: I think it's just the github diff, but the formatting on this line appears odd.

Comment (wojtek-t) [inline pkg/master/master_test.go]: I moved all arguments to the separate line - does it sound ok?

File: pkg/master/master_test.go [hunk 2]
```diff
@@ -121,27 +125,30 @@ func TestNew(t *testing.T) {
...
 // TestGetServersToValidate verifies the unexported getServersToValidate function
 func TestGetServersToValidate(t *testing.T) {
-	master, config, assert := setUp(t)
+	master, etcdserver, config, assert := setUp(t)
+	defer etcdserver.Terminate(t)
+
 	servers := master.getServersToValidate(&config)
 
-	assert.Equal(5, len(servers), "unexpected server list: %#v", servers)
+	assert.Equal(3, len(servers), "unexpected server list: %#v", servers)
```
Comment (timothysc) [inline pkg/master/master_test.go]: perhaps a comment on why 3.

Comment (wojtek-t) [inline pkg/master/master_test.go]: Done.

**Comment by wojtek-t** [comment]
@timothysc - PTAL

**Comment by timothysc** [comment]
Very minor comment on magic numbers otherwise lgtm.

**Comment by timothysc** [comment]
lgtm, pending tests.

**Comment by wojtek-t** [comment]
@timothysc - thanks (I added a comment about this magic number)
I'm applying "lgtm" labels based on your "LGTM" above.

**Pull request merged**